clarity_ext/extension_context.py

```diff
@@ -156,6 +144,48 @@ def cleanup(self):
                 os.remove(path)
 
 
+class MatchedAnalytes:
```

Comment:
It seems that it's not necessary to have a class for encapsulating this. I would rather just create helper methods/functions to use in the context object.